### ISSUE TITLE
parse_content_stream not returning all contents of a page

### ISSUE THREAD

**Issue by rain01** [pikepdf/pikepdf#230]
I am trying to access all operands and operators that make up the whole contents of the pdf and usually I can see all contents with this:

```
    for operands, operator in parse_content_stream(page):
        print(f"Operands {operands}, operator {operator}")
```
This usually returns hundreds if not thousands of commands, all the contents, but for one pdf I only get this:

```
Operands [], operator q
Operands [], operator q
Operands [Decimal('1.00'), Decimal('0.00'), Decimal('0.00'), Decimal('1.00'), Decimal('0.00'), Decimal('792.00')], operator cm
Operands [Decimal('0.00'), Decimal('0.00'), Decimal('612.00'), Decimal('-792.00')], operator re
Operands [], operator h
Operands [], operator W
Operands [], operator n
Operands [Decimal('1.00'), Decimal('0.00'), Decimal('0.00'), Decimal('1.00'), Decimal('0.00'), Decimal('0.00')], operator cm
Operands [], operator q
Operands [Decimal('1.00'), Decimal('0.00'), Decimal('0.00'), Decimal('1.00'), Decimal('0.00'), Decimal('-792.00')], operator cm
Operands [pikepdf.Name("/8b665375-51db-4ee1-8b3e-d4b85494bd70")], operator Do
Operands [], operator Q
Operands [], operator Q
Operands [], operator Q
Operands [], operator q
Operands [pikepdf.Name("/X1")], operator Do
Operands [], operator Q
```
While this page actually has multiple paragraphs of text, none of it appears in the content stream.

I am guessing all contents are coming from `Do` operator `/8b665375-51db-4ee1-8b3e-d4b85494bd70` but how can I access these contents?   
`Page` -> `/Resources` -> `/XObject` has `/8b665375-51db-4ee1-8b3e-d4b85494bd70` but I still can't figure out how to read or add/remove any contents in that xobject.

If it helps, the pdf was created using PDFKit.NET 5.0.28.0

Would it be possible to "normalize" this pdf to have all contents accessible with `parse_content_stream(page)`?

**Comment by jbarlow83** [comment]
`parse_content_stream` does do what it's suppose to here - you're getting a full representation of the objects on that page. One of them happens to be a Form XObject, and these can be shared by multiple pages or other Form XObjects.

When parsing the content stream, you'd want to save the name of any object you found that was referenced by a `Do` operator, look it up in the resources dictionary, and then run `parse_content_stream` on (in your case):
`parse_content_stream(Page.Resources.XObject['/8b665375-51db-4ee1-8b3e-d4b85494bd70'])`

----

As for whether to do so some sort of normalization. This may involve too much PDF nerd detail but I'll lay out why it is the way it is currently for posterity. 

Flattening into the main content stream is not necessarily desirable because they can be repeated content. Whether `parse_content_stream` should optional traverse into formxs without reformatting - that's a tough call, but this is a situation where I don't think it's wise for pikepdf to try to hide the underlying complexity of PDF. Most of the problems have to do with the fact that the formx has its own resources dictionary, and merging into it the page means merging the resource dictionary and rewriting the formx's content stream. A parser shouldn't change the data it's parsing.

There are also some complex issues with state when injecting a formx into a regular content stream.

**Comment by rain01** [comment]
Thank you for the quick response and explaining little about what is happening in the background there.  
There is no denying that PDFs are complex but I need some sort of way to access the contents of that pdf to work with it.

For example, I ran `parse_content_stream(Page.Resources.XObject['/384ee606-c7dd-4e80-8899-166463e91e97'])` and it returned another nested xobject/`Do` operator:

```
Operands [], operator q
Operands [Decimal('1.00'), Decimal('0.00'), Decimal('0.00'), Decimal('1.00'), Decimal('0.00'), Decimal('792.00')], operator cm
Operands [Decimal('0.00'), Decimal('0.00'), Decimal('612.00'), Decimal('-792.00')], operator re
Operands [], operator h
Operands [], operator W
Operands [], operator n
Operands [Decimal('1.00'), Decimal('0.00'), Decimal('0.00'), Decimal('1.00'), Decimal('0.00'), Decimal('0.00')], operator cm
Operands [], operator q
Operands [Decimal('0.00'), Decimal('0.00'), Decimal('612.00'), Decimal('-792.00')], operator re
Operands [], operator W
Operands [], operator n
Operands [], operator q
Operands [Decimal('0.75'), Decimal('0.00'), Decimal('0.00'), Decimal('0.75'), Decimal('0.00'), Decimal('-792.00')], operator cm
Operands [pikepdf.Name("/eb854fe4-07d5-4546-b7d4-2bfb3229251d")], operator Do
Operands [], operator Q
Operands [], operator Q
Operands [], operator Q
```

Printing out the Page I see it keeps going few layers deep of xobjects:

```
<pikepdf.Page({
  "/Contents": pikepdf.Stream(stream_dict = {
    "/Length": 303
  }, data = < ... > ),
  "/MediaBox": [Decimal('0.00000'), Decimal('0.00000'), Decimal('612.00000'), Decimal('792.00000')],
  "/Parent": < reference to / Pages > ,
  "/Resources": {
    "/ProcSet": ["/PDF", "/ImageB", "/ImageC", "/ImageI", "/Text"],
    "/XObject": {
      "/384ee606-c7dd-4e80-8899-166463e91e97": pikepdf.Stream(stream_dict = {
        "/BBox": [Decimal('0.00000'), Decimal('0.00000'), Decimal('612.00000'), Decimal('792.00000')],
        "/Filter": "/FlateDecode",
        "/Length": 176,
        "/Resources": {
          "/ProcSet": ["/PDF", "/ImageB", "/ImageC", "/ImageI", "/Text"],
          "/XObject": {
            "/eb854fe4-07d5-4546-b7d4-2bfb3229251d": pikepdf.Stream(stream_dict = {
              "/BBox": [Decimal('0.00000'), Decimal('0.00000'), Decimal('816.00000'), Decimal('1056.00000')],
              "/Filter": "/FlateDecode",
              "/Length": 11991,
              "/Resources": {
                "/ExtGState": {
                  "/8d0da6a9-7c2d-43c6-95bc-68d00db3c8da": {
                    "/BM": "/Normal",
                    "/CA": Decimal('1.00000'),
                    "/ca": Decimal('1.00000')
                  }
                },
                "/Font": {
                  "/5e92e14a-fcb4-4e48-a75b-ac7e62c32354": {
                    "/BaseFont": "/ZUDKKQ+Times New Roman",
                    "/DescendantFonts": [{
                      "/BaseFont": "/ZUDKKQ+Times New Roman",
                      "/CIDSystemInfo": {
                        "/Ordering": "Identity",
                        "/Registry": "Adobe",
                        "/Supplement": 0
                      },
                      "/CIDToGIDMap": "/Identity",
                      "/DW": 1000,
                      "/FontDescriptor": {
                        "/Ascent": 693,
                        "/AvgWidth": 250,
                        "/CIDSet": pikepdf.Stream(stream_dict = {
                          "/Filter": "/FlateDecode",
                          "/Length": 15
                        }, data = < ... > ),
                        "/CapHeight": 1356,
                        "/Descent": -215,
                        "/Flags": 32,
                        "/FontBBox": [Decimal('-568.00000'), Decimal('-216.00000'), Decimal('2045.00000'), Decimal('935.00000')],
                        "/FontFile2": pikepdf.Stream(stream_dict = {
                          "/Filter": "/FlateDecode",
                          "/Length": 14255
                        }, data = < ... > ),
                        "/FontName": "/ZUDKKQ+Times New Roman",
                        "/ItalicAngle": 0,
                        "/Leading": 1059,
                        "/MaxWidth": 250,
                        "/MissingWidth": 250,
                        "/StemH": 0,
                        "/StemV": 80,
                        "/Type": "/FontDescriptor",
                        "/XHeight": 0
                      },
                      "/Subtype": "/CIDFontType2",
                      "/Type": "/Font",
                      "/W": [179, [443, 443], 182, [333]]
                    }],
                    "/Encoding": "/Identity-H",
                    "/Subtype": "/Type0",
                    "/ToUnicode": pikepdf.Stream(stream_dict = {
                      "/Filter": "/FlateDecode",
                      "/Length": 245
                    }, data = < ... > ),
                    "/Type": "/Font"
                  },
                  "/846e6a21-e7a3-4f75-90b2-2d14454892ba": {
                    "/BaseFont": "/TimesNewRomanPSMT",
                    "/Encoding": "/WinAnsiEncoding",
                    "/FirstChar": 30,
                    "/FontDescriptor": {
                      "/Ascent": 693,
                      "/AvgWidth": 250,
                      "/CapHeight": 500,
                      "/Descent": -216,
                      "/Flags": 34,
                      "/FontBBox": [Decimal('-568.00000'), Decimal('-216.00000'), Decimal('2046.00000'), Decimal('933.00000')],
                      "/FontName": "/TimesNewRomanPSMT",
                      "/ItalicAngle": 0,
                      "/Leading": 1149,
                      "/MaxWidth": 2614,
                      "/MissingWidth": 250,
                      "/StemH": 72,
                      "/StemV": 72,
                      "/Type": "/FontDescriptor",
                      "/XHeight": 250
                    },
                    "/LastChar": 255,
                    "/Subtype": "/TrueType",
                    "/Type": "/Font",
                    "/Widths": [778, 778, 250, 333, 408, 500, 500, 833, 778, 180, 333, 333, 500, 564, 250, 333, 250, 278, 500, 500, 500, 500, 500, 500, 500, 500, 500, 500, 278, 278, 564, 564, 564, 444, 921, 722, 667, 667, 722, 611, 556, 722, 722, 333, 389, 722, 611, 889, 722, 722, 556, 722, 667, 556, 611, 722, 722, 944, 722, 722, 611, 333, 278, 333, 469, 500, 333, 444, 500, 444, 500, 444, 333, 500, 500, 278, 278, 500, 278, 778, 500, 500, 500, 500, 333, 389, 278, 500, 500, 722, 500, 500, 444, 480, 200, 480, 541, 778, 500, 778, 333, 500, 444, 1000, 500, 500, 333, 1000, 556, 333, 889, 778, 611, 778, 778, 333, 333, 444, 444, 350, 500, 1000, 333, 980, 389, 333, 722, 778, 444, 722, 250, 333, 500, 500, 500, 500, 200, 500, 333, 760, 276, 500, 564, 333, 760, 500, 400, 549, 300, 300, 333, 576, 453, 333, 333, 300, 310, 500, 750, 750, 750, 444, 722, 722, 722, 722, 722, 722, 889, 667, 611, 611, 611, 611, 333, 333, 333, 333, 722, 722, 722, 722, 722, 722, 722, 564, 722, 722, 722, 722, 722, 722, 556, 500, 444, 444, 444, 444, 444, 444, 667, 444, 444, 444, 444, 444, 278, 278, 278, 278, 500, 500, 500, 500, 500, 500, 500, 549, 500, 500, 500, 500, 500, 500, 500, 500]
                  },
                  "/95f11355-ead5-4b59-ae7d-c9bc0af7bf1d": {
                    "/BaseFont": "/TimesNewRomanPS-BoldMT",
                    "/Encoding": "/WinAnsiEncoding",
                    "/FirstChar": 30,
                    "/FontDescriptor": {
                      "/Ascent": 677,
                      "/AvgWidth": 250,
                      "/CapHeight": 500,
                      "/Descent": -216,
                      "/Flags": 34,
                      "/FontBBox": [Decimal('-558.00000'), Decimal('-216.00000'), Decimal('2000.00000'), Decimal('933.00000')],
                      "/FontName": "/TimesNewRomanPS-BoldMT",
                      "/ItalicAngle": 0,
                      "/Leading": 1149,
                      "/MaxWidth": 2558,
                      "/MissingWidth": 250,
                      "/StemH": 144,
                      "/StemV": 144,
                      "/Type": "/FontDescriptor",
                      "/XHeight": 250
                    },
                    "/LastChar": 255,
                    "/Subtype": "/TrueType",
                    "/Type": "/Font",
                    "/Widths": [778, 778, 250, 333, 555, 500, 500, 1000, 833, 278, 333, 333, 500, 570, 250, 333, 250, 278, 500, 500, 500, 500, 500, 500, 500, 500, 500, 500, 333, 333, 570, 570, 570, 500, 930, 722, 667, 722, 722, 667, 611, 778, 778, 389, 500, 778, 667, 944, 722, 778, 611, 778, 722, 556, 667, 722, 722, 1000, 722, 722, 667, 333, 278, 333, 581, 500, 333, 500, 556, 444, 556, 444, 333, 500, 556, 278, 333, 556, 278, 833, 556, 500, 556, 556, 444, 389, 333, 556, 500, 722, 500, 500, 444, 394, 220, 394, 520, 778, 500, 778, 333, 500, 500, 1000, 500, 500, 333, 1000, 556, 333, 1000, 778, 667, 778, 778, 333, 333, 500, 500, 350, 500, 1000, 333, 1000, 389, 333, 722, 778, 444, 722, 250, 333, 500, 500, 500, 500, 220, 500, 333, 747, 300, 500, 570, 333, 747, 500, 400, 549, 300, 300, 333, 576, 540, 333, 333, 300, 330, 500, 750, 750, 750, 500, 722, 722, 722, 722, 722, 722, 1000, 722, 667, 667, 667, 667, 389, 389, 389, 389, 722, 722, 778, 778, 778, 778, 778, 570, 778, 722, 722, 722, 722, 722, 611, 556, 500, 500, 500, 500, 500, 500, 722, 444, 444, 444, 444, 444, 278, 278, 278, 278, 500, 556, 500, 500, 500, 500, 500, 549, 500, 556, 556, 556, 556, 500, 556, 500]
                  },
                  "/e462b664-b798-4523-ad6a-a6c4722711e0": {
                    "/BaseFont": "/HMMBWW+Times New Roman Bold Italic",
                    "/DescendantFonts": [{
                      "/BaseFont": "/HMMBWW+Times New Roman Bold Italic",
                      "/CIDSystemInfo": {
                        "/Ordering": "Identity",
                        "/Registry": "Adobe",
                        "/Supplement": 0
                      },
                      "/CIDToGIDMap": "/Identity",
                      "/DW": 1000,
                      "/FontDescriptor": {
                        "/Ascent": 677,
                        "/AvgWidth": 250,
                        "/CIDSet": pikepdf.Stream(stream_dict = {
                          "/Filter": "/FlateDecode",
                          "/Length": 14
                        }, data = < ... > ),
                        "/CapHeight": 1356,
                        "/Descent": -215,
                        "/Flags": 262240,
                        "/FontBBox": [Decimal('-547.00000'), Decimal('-216.00000'), Decimal('1400.00000'), Decimal('935.00000')],
                        "/FontFile2": pikepdf.Stream(stream_dict = {
                          "/Filter": "/FlateDecode",
                          "/Length": 11524
                        }, data = < ... > ),
                        "/FontName": "/HMMBWW+Times New Roman Bold Italic",
                        "/ItalicAngle": -18,
                        "/Leading": 1042,
                        "/MaxWidth": 250,
                        "/MissingWidth": 250,
                        "/StemH": 0,
                        "/StemV": 80,
                        "/Type": "/FontDescriptor",
                        "/XHeight": 0
                      },
                      "/Subtype": "/CIDFontType2",
                      "/Type": "/Font",
                      "/W": [182, [333]]
                    }],
                    "/Encoding": "/Identity-H",
                    "/Subtype": "/Type0",
                    "/ToUnicode": pikepdf.Stream(stream_dict = {
                      "/Filter": "/FlateDecode",
                      "/Length": 227
                    }, data = < ... > ),
                    "/Type": "/Font"
                  },
                  "/f8a85476-c5c9-4e6d-9286-c43a46fef982": {
                    "/BaseFont": "/TimesNewRomanPS-BoldItalicMT",
                    "/Encoding": "/WinAnsiEncoding",
                    "/FirstChar": 30,
                    "/FontDescriptor": {
                      "/Ascent": 677,
                      "/AvgWidth": 250,
                      "/CapHeight": 500,
                      "/Descent": -216,
                      "/Flags": 98,
                      "/FontBBox": [Decimal('-547.00000'), Decimal('-216.00000'), Decimal('1401.00000'), Decimal('933.00000')],
                      "/FontName": "/TimesNewRomanPS-BoldItalicMT",
                      "/ItalicAngle": -16,
                      "/Leading": 1149,
                      "/MaxWidth": 1948,
                      "/MissingWidth": 250,
                      "/StemH": 144,
                      "/StemV": 144,
                      "/Type": "/FontDescriptor",
                      "/XHeight": 250
                    },
                    "/LastChar": 255,
                    "/Subtype": "/TrueType",
                    "/Type": "/Font",
                    "/Widths": [778, 778, 250, 389, 555, 500, 500, 833, 778, 278, 333, 333, 500, 570, 250, 333, 250, 278, 500, 500, 500, 500, 500, 500, 500, 500, 500, 500, 333, 333, 570, 570, 570, 500, 832, 667, 667, 667, 722, 667, 667, 722, 778, 389, 500, 667, 611, 889, 722, 722, 611, 722, 667, 556, 611, 722, 667, 889, 667, 611, 611, 333, 278, 333, 570, 500, 333, 500, 500, 444, 500, 444, 333, 500, 556, 278, 278, 500, 278, 778, 556, 500, 500, 500, 389, 389, 278, 556, 444, 667, 500, 444, 389, 348, 220, 348, 570, 778, 500, 778, 333, 500, 500, 1000, 500, 500, 333, 1000, 556, 333, 944, 778, 611, 778, 778, 333, 333, 500, 500, 350, 500, 1000, 333, 1000, 389, 333, 722, 778, 389, 611, 250, 389, 500, 500, 500, 500, 220, 500, 333, 747, 266, 500, 606, 333, 747, 500, 400, 549, 300, 300, 333, 576, 500, 250, 333, 300, 300, 500, 750, 750, 750, 500, 667, 667, 667, 667, 667, 667, 944, 667, 667, 667, 667, 667, 389, 389, 389, 389, 722, 722, 722, 722, 722, 722, 722, 570, 722, 722, 722, 722, 722, 611, 611, 500, 500, 500, 500, 500, 500, 500, 722, 444, 444, 444, 444, 444, 278, 278, 278, 278, 500, 556, 500, 500, 500, 500, 500, 549, 500, 556, 556, 556, 556, 444, 500, 444]
                  }
                },
                "/ProcSet": ["/PDF", "/ImageB", "/ImageC", "/ImageI", "/Text"],
                "/XObject": {
                  "/77dcd0c4-df27-47d5-a28f-4c1421269f7e": pikepdf.Stream(stream_dict = {
                    "/BitsPerComponent": 8,
                    "/ColorSpace": "/DeviceRGB",
                    "/Decode": [Decimal('0.00000'), Decimal('1.00000'), Decimal('0.00000'), Decimal('1.00000'), Decimal('0.00000'), Decimal('1.00000')],
                    "/DecodeParms": {
                      "/BlackIs1": True,
                      "/Columns": 2550,
                      "/K": -1,
                      "/Rows": 256
                    },
                    "/Filter": "/DCTDecode",
                    "/Height": 256,
                    "/Length": 28713,
                    "/SMask": pikepdf.Stream(stream_dict = {
                      "/BitsPerComponent": 8,
                      "/ColorSpace": "/DeviceGray",
                      "/Filter": "/FlateDecode",
                      "/Height": 256,
                      "/Length": 5457,
                      "/Matte": [0, 0, 0],
                      "/Subtype": "/Image",
                      "/Type": "/XObject",
                      "/Width": 2550
                    }, data = < ... > ),
                    "/Subtype": "/Image",
                    "/Type": "/XObject",
                    "/Width": 2550
                  }, data = < ... > ),
                  "/b70931e4-b943-460c-8807-f90b6a5b896c": pikepdf.Stream(stream_dict = {
                    "/BitsPerComponent": 8,
                    "/ColorSpace": "/DeviceRGB",
                    "/Decode": [Decimal('0.00000'), Decimal('1.00000'), Decimal('0.00000'), Decimal('1.00000'), Decimal('0.00000'), Decimal('1.00000')],
                    "/DecodeParms": {
                      "/BlackIs1": True,
                      "/Columns": 2550,
                      "/K": -1,
                      "/Rows": 256
                    },
                    "/Filter": "/DCTDecode",
                    "/Height": 256,
                    "/Length": 28713,
                    "/SMask": pikepdf.Stream(stream_dict = {
                      "/BitsPerComponent": 8,
                      "/ColorSpace": "/DeviceGray",
                      "/Filter": "/FlateDecode",
                      "/Height": 256,
                      "/Length": 5457,
                      "/Matte": [0, 0, 0],
                      "/Subtype": "/Image",
                      "/Type": "/XObject",
                      "/Width": 2550
                    }, data = < ... > ),
                    "/Subtype": "/Image",
                    "/Type": "/XObject",
                    "/Width": 2550
                  }, data = < ... > ),
                  "/c3c1963e-0ef8-495e-be5a-4879e8a8448c": pikepdf.Stream(stream_dict = {
                    "/BitsPerComponent": 8,
                    "/ColorSpace": "/DeviceRGB",
                    "/Decode": [Decimal('0.00000'), Decimal('1.00000'), Decimal('0.00000'), Decimal('1.00000'), Decimal('0.00000'), Decimal('1.00000')],
                    "/DecodeParms": {
                      "/BlackIs1": True,
                      "/Columns": 2550,
                      "/K": -1,
                      "/Rows": 516
                    },
                    "/Filter": "/DCTDecode",
                    "/Height": 516,
                    "/Length": 49191,
                    "/SMask": pikepdf.Stream(stream_dict = {
                      "/BitsPerComponent": 8,
                      "/ColorSpace": "/DeviceGray",
                      "/Filter": "/FlateDecode",
                      "/Height": 516,
                      "/Length": 16561,
                      "/Matte": [0, 0, 0],
                      "/Subtype": "/Image",
                      "/Type": "/XObject",
                      "/Width": 2550
                    }, data = < ... > ),
                    "/Subtype": "/Image",
                    "/Type": "/XObject",
                    "/Width": 2550
                  }, data = < ... > )
                }
              },
              "/Subtype": "/Form",
              "/Type": "/XObject"
            }, data = < ... > )
          }
        },
        "/Subtype": "/Form",
        "/Type": "/XObject"
      }, data = < ... > ),
      "/X1": pikepdf.Stream(stream_dict = {
        "/BBox": [Decimal('0.00000'), Decimal('0.00000'), Decimal('612.00000'), Decimal('792.00000')],
        "/Filter": "/FlateDecode",
        "/FormType": 1,
        "/Length": 300,
        "/Resources": {
          "/ExtGState": {
            "/G0": {
              "/AIS": False,
              "/CA": Decimal('0.25000'),
              "/Type": "/ExtGState",
              "/ca": Decimal('0.25000')
            },
            "/G1": {
              "/AIS": False,
              "/CA": Decimal('1.00000'),
              "/Type": "/ExtGState",
              "/ca": Decimal('1.00000')
            },
            "/G2": {
              "/AIS": False,
              "/CA": Decimal('0.25000'),
              "/Type": "/ExtGState",
              "/ca": Decimal('0.25000')
            },
            "/G3": {
              "/AIS": False,
              "/CA": Decimal('1.00000'),
              "/Type": "/ExtGState",
              "/ca": Decimal('1.00000')
            },
            "/G4": {
              "/AIS": False,
              "/CA": Decimal('0.25000'),
              "/Type": "/ExtGState",
              "/ca": Decimal('0.25000')
            },
            "/G5": {
              "/AIS": False,
              "/CA": Decimal('1.00000'),
              "/Type": "/ExtGState",
              "/ca": Decimal('1.00000')
            }
          },
          "/Font": {
            "/tallF4": {
              "/BaseFont": "/Helvetica",
              "/Encoding": "/WinAnsiEncoding",
              "/Name": "/tallF4",
              "/Subtype": "/Type1",
              "/Type": "/Font"
            }
          }
        },
        "/Subtype": "/Form",
        "/Type": "/XObject"
      }, data = < ... > )
    }
  },
  "/Type": "/Page"
})>
```

Is my only option to access any contents just recursing into each xobject to try and reach the contents(assuming the text content is at the last xobject)?  
I guess I was really hoping there was some magic that would simplify the complexity of this craziness.

**Comment by jbarlow83** [comment]
I agree the interface could be improved. What are actually trying to achieve?

I think if you use Ghostscript to rewrite ("re-fry") the PDF, it might flatten the Form XObjects for you. Something like `gs -q -sDEVICE=pdfwrite -o output.pdf input.pdf`. That may be a suitable workaround.

**Comment by rain01** [comment]
Ghostscript actually helped.  
I try to extract tables from pdf with Camelot but often times I need to clean up the contents a little before I run pdf through Camelot, otherwise sometimes I get messy results. eg. I have a list of strings I want to remove and then I try to remove them before I pass pdf to Camelot.
It is kind of similar to what [faysou posted here](https://stackoverflow.com/a/57410205/1537657) but using pikepdf instead of PyPDF.
Using `parse_content_stream()` has worked great for most of the pdfs but every now and then there are those that I haven't been able to clean up.  

My progress so far:
 - With `gs` it actually flattened the contents and now my clean-up method works on this same as rest of the pdfs.  
 - I built a recursive method that would go inside each xobjects content stream, I want see if I can clean it that way also without needing `gs` to re-fry the pdf to not depend on command line program and as I understand re-frying may come with a risk of data loss or something going wrong.

Would be great if I could do it all with just pikepdf.

**Comment by jbarlow83** [comment]
This may have been fixed in commit 4db699ea for v5.3.1
